### PR TITLE
[ISSUE #4029] add  shenyu-client-api-doc

### PR DESCRIPTION
**Pull request by 847850277**
Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java [hunk 1]
```diff
@@ -175,4 +175,25 @@ public CommonPager<ApiVO> listByPage(final ApiQuery apiQuery) {
...
+                tagRelationMapper.deleteByApiIds(apiIds);
+            }
+            return deleteRows;
+        }
+        return 0;
+    }
+
+    @Override
+    public String offlineByContextPath(final String contextPath) {
+        apiMapper.updateOfflineByContextPath(contextPath);
```
Comment (dragon-zhang) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/ApiServiceImpl.java:196]: There are many nodes in the service. Only when all nodes are offline can the API status be updated.

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractShenyuClientRegisterServiceImpl.java [hunk 2]
```diff
@@ -54,53 +60,58 @@
...
     @Resource
     private MetaDataService metaDataService;
-    
+
     /**
      * The Rule service.
      */
     @Resource
     private RuleService ruleService;
-
```
Comment (dragon-zhang) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractShenyuClientRegisterServiceImpl.java:78]: do not change the format, try config format in your IDE.

**Comment by dragon-zhang** [comment]
There are too many files changed. Please split them into at least 3 PRs, and complete client, example, and admin respectively.

**Review by dragon-zhang — CHANGES_REQUESTED** [review]
Please split this PR into at least 3 PRs.

**Comment by 847850277** [comment]
> Please split this PR into at least 3 PRs.

It will be solved when I'm empty.

**Pull request closed without merging**